File: scripts/test-items-api.ts
#!/usr/bin/env npx tsx
import fetch from 'node-fetch';

const BASE_URL = 'https://jobeye-production.up.railway.app';
// const BASE_URL = 'http://localhost:3000'; // For local testing

async function testDebugEndpoint() {
  console.log('🧪 Testing Items Debug Endpoint...\n');

  // Test GET
  console.log('1️⃣ Testing GET /api/debug/items');
  try {
    const getResponse = await fetch(`${BASE_URL}/api/debug/items`, {
      headers: {
        'Accept': 'application/json'
      }
    });
    const getData = await getResponse.json();
    console.log('Status:', getResponse.status);
    console.log('Response:', JSON.stringify(getData, null, 2));
  } catch (error) {
    console.error('GET Error:', error);
  }

  console.log('\n2️⃣ Testing POST /api/debug/items');
  try {
    const postResponse = await fetch(`${BASE_URL}/api/debug/items`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json'
      },
      body: JSON.stringify({
        name: 'Test Lawn Mower',
        item_type: 'equipment',
        category: 'lawn-care',
        tracking_mode: 'individual',
        current_quantity: 1,
        unit_of_measure: 'each',
        description: 'Test from API script'
      })
    });
    const postData = await postResponse.json();
    console.log('Status:', postResponse.status);
    console.log('Response:', JSON.stringify(postData, null, 2));
  } catch (error) {
    console.error('POST Error:', error);
  }
}

async function testItemsAPI() {
  console.log('\n\n🎯 Testing Items API...\n');

  // Test GET items
  console.log('3️⃣ Testing GET /api/supervisor/items');
  try {
    const response = await fetch(`${BASE_URL}/api/supervisor/items`, {
      headers: {
        'Accept': 'application/json',
        'x-tenant-id': '00000000-0000-0000-0000-000000000000'
      }
    });
    const data = await response.json();
    console.log('Status:', response.status);
    console.log('Response:', JSON.stringify(data, null, 2));
  } catch (error) {
    console.error('Error:', error);
  }

  // Test POST item
  console.log('\n4️⃣ Testing POST /api/supervisor/items');
  try {
    const response = await fetch(`${BASE_URL}/api/supervisor/items`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'x-tenant-id': '00000000-0000-0000-0000-000000000000'
      },
      body: JSON.stringify({
        item_type: 'material',
        category: 'lawn-care',
        name: 'Grass Seed',
        tracking_mode: 'quantity',
        current_quantity: 25,
        unit_of_measure: 'pound',
        min_quantity: 10,
        reorder_point: 15,
        description: 'Premium grass seed mix'
      })
    });
    const data = await response.json();
    console.log('Status:', response.status);
    console.log('Response:', JSON.stringify(data, null, 2));
    
    // If successful, test GET single item
    if (response.status === 201 && data.item?.id) {
      console.log('\n5️⃣ Testing GET /api/supervisor/items/[id]');
      const getOne = await fetch(`${BASE_URL}/api/supervisor/items/${data.item.id}`, {
        headers: {
          'Accept': 'application/json',
          'x-tenant-id': '00000000-0000-0000-0000-000000000000'
        }
      });
      const oneData = await getOne.json();
      console.log('Status:', getOne.status);
      console.log('Response:', JSON.stringify(oneData, null, 2));
    }
  } catch (error) {
    console.error('Error:', error);
  }
}

// Run tests
(async () => {
  await testDebugEndpoint();
  await testItemsAPI();
})();

File: scripts/test-items-api.py
#!/usr/bin/env python3
import requests
import os
from dotenv import load_dotenv

# Load environment variables
load_dotenv(dotenv_path='.env.local')

SUPABASE_URL = os.getenv('NEXT_PUBLIC_SUPABASE_URL')

print("🔍 Testing items API endpoint...\n")

# Test the production items API
production_url = "https://jobeye-production.up.railway.app"
local_url = "http://localhost:3000"

# Try production first
print("1. Testing production API:")
try:
    response = requests.get(
        f"{production_url}/api/supervisor/items",
        headers={
            'x-tenant-id': 'demo-company'
        }
    )
    print(f"   Status: {response.status_code}")
    if response.status_code == 200:
        data = response.json()
        print(f"   Response keys: {list(data.keys())}")
        if 'items' in data:
            print(f"   Items count: {len(data['items'])}")
            if data['items']:
                print(f"   First item: {data['items'][0]}")
    else:
        print(f"   Response: {response.text[:200]}")
except Exception as e:
    print(f"   Error: {e}")

# Test direct Supabase query
print("\n2. Testing direct Supabase query:")
headers = {
    "apikey": os.getenv('SUPABASE_SERVICE_ROLE_KEY'),
    "Authorization": f"Bearer {os.getenv('SUPABASE_SERVICE_ROLE_KEY')}",
    "Content-Type": "application/json"
}

response = requests.get(
    f"{SUPABASE_URL}/rest/v1/items?tenant_id=eq.demo-company",
    headers=headers
)
print(f"   Status: {response.status_code}")
if response.status_code == 200:
    items = response.json()
    print(f"   Items found: {len(items)}")
    if items:
        print(f"   Sample item: {items[0]}")
else:
    print(f"   Error: {response.text}")

# Check if 'demo-company' tenant exists
print("\n3. Checking all unique tenant_ids in items:")
response = requests.get(
    f"{SUPABASE_URL}/rest/v1/items?select=tenant_id&limit=100",
    headers=headers
)
if response.status_code == 200:
    items = response.json()
    tenant_ids = list(set([item['tenant_id'] for item in items]))
    print(f"   Unique tenant_ids: {tenant_ids}")